docs/using-react-redux/connect-dispatching-actions-with-mapDispatchToProps.md

```diff
@@ -0,0 +1,419 @@
+---
+id: connect-dispatching-actions-with-mapDispatchToProps
+title: Connect: Dispatching Actions with mapDispatchToProps
+hide_title: true
+sidebar_label: Connect: Dispatching Actions with mapStateToProps
+---
+
+# Connect: Dispatching Actions with `mapDispatchToProps`
+
+As the second argument passed in to `connect`, `mapDispatchToProps` is used for dispatching actions to the store.
+
+`dispatch` is a function of the Redux store. You call `store.dispatch` to dispatch an action.
+This is the only way to trigger a state change.
+
+With React-Redux, your components never access the store directly - `connect` does it for you.
+React-Redux gives you two ways to let components dispatch actions:
+
+- By default, a connected component receives `props.dispatch` and can dispatch actions itself.
+- `connect` can accept an argument called `mapDispatchToProps`, which lets you create functions that dispatch when called, and pass those functions as props to your component.
+
+The `mapDispatchToProps` functions are normally referred to as `mapDispatch` for short, and of course the actual variable name doesn’t matter in real code.
+
+## With or Without `mapDispatchToProps`
+
+### Default: `dispatch` as a Prop
+
+If you don't specify the second argument to `connect()`, your component will receive `dispatch` by default. For example:
+
+```js
+connect()(MyComponent);
+// which is equivalent with
+connect(
+  null,
+  null
+)(MyComponent);
+
+// or
+connect(mapStateToProps /** no second argument */)(MyComponent);
+```
+
+Once you have connected your component in this way, your component receives `props.dispatch`. You may use it to dispatch actions to the store.
+
+```js
+function Counter({ count, dispatch }) {
+  return (
+    <div>
+      <button onClick={() => dispatch({ type: "DECREMENT" })}>-</button>
+      <span>{count}</span>
+      <button onClick={() => dispatch({ type: "INCREMENT" })}>+</button>
+      <button onClick={() => dispatch({ type: "RESET" })}>reset</button>
+    </div>
+  );
+}
+```
+
+### Providing A `mapDispatchToProps`
+
+Providing a `mapDispatchToProps` allows you to specify which actions your component might need to dispatch. It lets you provide action dispatching functions as props. Therefore, instead of calling `props.dispatch(() => increment())`, you may call `props.increment()` directly. There are a few reasons why you might want to do that.
+
+#### More Declarative
+
+First, encapsulating the dispatch logic into function makes the implementation more declarative.
+Dispatching an action and let the Redux store handle the data flow is _how to_ implement the feature, it is not _what_ it does.
```

Comment:
```suggestion
Dispatching an action and letting the Redux store handle the data flow is _how to_ implement the behavior, rather than _what_ it does.
```